djaoapp/api/contact.py

```diff
@@ -111,3 +115,118 @@ def post(self, request, *args, **kwargs):
 _("Sorry, there was an issue sending your request for information"\
 " to '%(full_name)s &lt;%(email)s&gt;'.") % {
     'full_name': provider.full_name, 'email': provider.email}})
+
+
+class TypeaheadQueryAPIView(GenericAPIView):
+    """
+    Returns address typeahead suggestions
+
+    The API is typically used within a profile page.
+
+    **Tags**: profile
+
+    **Examples
+
+    .. code-block:: http
+
+        GET /api/typeahead/q?q=123 HTTP/1.1
+
+    responds
+
+    .. code-block:: json
+
+       {
+            "count": 5,
+            "results": [
+                {
+                "description": "123 William Street, New York, NY, USA",
+                "place_id": "ChIJIaGbBBhawokRUmbgNsUmr-s"
+                },
+                {
+                "description": "1230 6th Avenue, New York, NY, USA",
+                "place_id": "ChIJC8abO_9YwokRlbzMgC8_XWE"
+                },
+                {
+                "description": "1233 York Avenue, New York, NY, USA",
+                "place_id": "ChIJ1_GF_cJYwokR8u455JHGkBA"
+                },
+                {
+                "description": "123 Melrose Street, Brooklyn, NY, USA",
+                "place_id": "ChIJZbhuygdcwokR_5NT5POb8Z8"
+                },
+                {
+                "description": "1239 Broadway, New York, NY, USA",
+                "place_id": "ChIJMebitahZwokRa9MwUrkbedc"
+                }
+            ]
+        }
+    """
+    serializer_class = TypeaheadSuggestionSerializer
+
+    @swagger_auto_schema(responses={
+        200: OpenAPIResponse("success", ValidationErrorSerializer)})
+    def get(self, request, *args, **kwargs):
+        gmaps = googlemaps.Client(key=settings.GOOGLE_API_KEY)
+        query = request.GET.get('q')
```

Comment:
Hmm, I'm not sure I understand: we're not paginating here, the `2` is the minimal required length for the query (to avoid extra API call to Google Maps).